app/jinja_filters.py

```diff
@@ -52,6 +52,31 @@ def format_percentage(value):
     return '{}%'.format(value)
 
 
+@blueprint.app_template_filter()
+def format_address_list(user_entered_address=None, metadata_address=None):
+    """
+        This function format_address_list accepts two addresses, a user submitted address and a metadata address
+        If all the items in address list are 'Undefined' (nothing in answer_store) then we retrieve the metadata.
+
+        :param user_entered_address user entered address which is a list of answer fields
+        :param metadata_address metadata address which is a list of answer fields passed to us
+        :return: the value of the address to be piped
+    """
+    if all(isinstance(field, Undefined) for field in user_entered_address):
+        address = metadata_address
+    elif all(field == '' for field in user_entered_address):
+        address = metadata_address
+    else:
```

Comment:
If you do find issues @jonnyshaw89 can you add that this `if and elif` combo should just be an `or` as it's doing the same thing.

Reply:
Yeah, I thought about that, but then thought it looks easier to read like this, happy to change

Reply:
Definitely better as an `or`. 
Otherwise you're using an if statement purely as a way to split up lines, which isn't what anyone would expect.